packages/react-reconciler/src/ReactFiber.js

```diff
@@ -417,7 +385,38 @@ export function createFiberFromElement(
   return fiber;
 }
 
-function throwOnInvalidElementType(type, owner) {
+function getFiberTagFromObjectType(type, owner): TypeOfWork {
+  if (typeof type !== 'object' || type === null) {
+    invariant(
```

Comment:
Had to inline these because Flow was getting confused when an error happened via another method.

Reply:
Do we need to declare this invariant twice? Did you consider something like this?
```js
function getFiberTagFromObjectType(type, owner): TypeOfWork {
  const $$typeof = typeof type === "object" && type !== null
    ? type.$$typeof
    : null;
  switch ($$typeof) {
    case REACT_PROVIDER_TYPE:
      return ContextProvider;
    case REACT_CONTEXT_TYPE:
      return ContextConsumer;
    case REACT_FORWARD_REF_TYPE:
      return ForwardRef;
    default:
      invariant(
        false,
        "Element type is invalid: expected a string (for built-in " +
          "components) or a class/function (for composite components) " +
          "but got: %s.%s",
        type == null ? type : typeof type,
        getInvalidElementTypeErrorInfo(type, owner)
      );
  }
}
```